corehq/apps/accounting/forms.py

```diff
@@ -566,7 +566,7 @@ def clean(self):
             and start_date > self.cleaned_data['end_date']):
             raise ValidationError("End date must be after start date.")
 
-        if self.cleaned_data['end_date'] <= datetime.date.today():
+        if self.cleaned_data['end_date'] and self.cleaned_data['end_date'] <= datetime.date.today():
```

Comment:
I know you didn't change this, but does it matter that `datetime.date.today()` is the server's local time? (also, according to @dannyroberts all of our servers' time zones are set to UTC, so not sure it matters practically right now, but it does seem sloppy and potentially problematic if that ever were not true)